nixos/tests/pangolin.nix

```
@@ -0,0 +1,134 @@
+{
+  lib,
+  pkgs,
+  ...
+}:
+let
+  # cant use .test, since that gets caught by traefik
+  domain = "nixos.eu";
+  newtId = "this.is.the.newt.id";
+  secret = "1234567890";
+
+  dnsServerIP = nodes: nodes.dnsserver.networking.primaryIPAddress;
+
+  dnsScript = pkgs.writeShellScript "dns-hook.sh" ''
+    set -euo pipefail
+    echo '[INFO]' "[$2]" 'dns-hook.sh' $*
+    if [ "$1" = "present" ]; then
+      ${pkgs.curl}/bin/curl --data @- http://dnsserver.eu:8055/set-txt << EOF
+      {"host": "$2", "value": "$3"}
+    EOF
+    else
+      ${pkgs.curl}/bin/curl --data @- http://dnsserver.eu:8055/clear-txt << EOF
+      {"host": "$2"}
+    EOF
+    fi
+  '';
+
+in
+{
+  name = "pangolin";
+  meta.maintainers = with lib.maintainers; [
+    jackr
+    sigmasquadron
+  ];
+
+  # a whole fleet of VM's:
+  # Panglin on the VPS, and Newt in the privateHost
+  # setup ACME server to sign certificates and point
+  # DNS to the correct machine
```

```suggestion
  # The full test is not yet implemented, but once upstream supports a way to
  # configure Pangolin non-interactively, the full test will look like the following:
  # - 'acme': ACME server to replace the real servers at Let's Encrypt.
  # - 'dnsserver': The pebble challenge test server so we can use a private DNS
  #                for everything here.
  # - 'VPS': The Pangolin instance, running Gerbil, Traefik, and Badger as well.
  # - 'privateHost': The private server running an HTTP server on its local
  #                  network that will be tunnelled via Newt to the VPS.
  # - 'client': An outside node that will test if the service hosted in
  #             'privateHost' is publicly accessible.
  # TODO: In the future, we should also have a machine to test the
  #       functionality of Olm, as well as a split Pangolin/Gerbil
  #       configuration once that is implemented into the module.
```
This should define a little more precisely what the test currently entails, and what it will entail in the future.